### DISCUSSION TITLE
Drawing a tree with subset of sample nodes and/or larger dots for sample nodes

### DISCUSSION THREAD

**Discussion by janaobsteter** [tskit-dev/tskit#2371] [Q&A] [answered]
I am trying to draw a nice tree and I'm failing.
This is my plot for a specific location in the honeybee genome. I have some samples with known subspecies (colours) and the majority of them are hybrids, which I've coloured in white. Is there any way to:
1. Draw a tree for a subset of the samples (just the coloured ones)
2. To increase the size of the sample nodes

The code I'm using at the moment is very simple:
```
ts10 = tskit.load("Chr10.trees")
yellowtree = ts10.at(11506337)
yellowtree.draw(
    path="yellowtreeStart.svg",
    height=700,
    width=1200,
    node_labels={},
    node_colours=colours_for_node,
)
```
Any help is much appreciated!

![image](https://user-images.githubusercontent.com/18226551/176641571-85bcd9e6-0f5b-4638-85d1-8c85e4dd8b2f.png)

**Comment by hyanwong** [comment, marked as answer]
For plotting a subset of the samples, you can use `simplify()`: there's an example in https://tskit.dev/tutorials/getting_started.html#processing-trees but you are quite right that it is a bit hidden, and we should also give an example in the "viz" tutorial (I've opened an issue: https://github.com/tskit-dev/tutorials/issues/182).

To increase the size of the nodes, you can use `draw_svg(symbol_size=5)` (pick whatever number seems right for your viz). If you only want to increase the size of the *sample* nodes, you could, for example, hide the other nodes by adding `".node:not(.leaf) > .sym {display: none}"` to the style (see https://tskit.dev/tutorials/viz.html#restricting-styling).

Note, however, that if you use the more sophisticated `draw_svg()` method you will need to colour the populations using css. There's an example in the [tutorial](https://tskit.dev/tutorials/viz.html#styling) and some discussion at https://github.com/tskit-dev/tskit/issues/2330#issuecomment-1151132640

**Reply by hyanwong** [reply]
Also note a little hack at https://github.com/tskit-dev/tskit/discussions/2365 which might make it easier to see clusters:

<img width="458" alt="Screenshot 2022-06-23 at 16 13 26" src="https://user-images.githubusercontent.com/4699014/175333463-07a1ef76-7db0-48fa-bab0-a0c9ba96aa41.png">

**Comment by janaobsteter** [comment]
Thanks! I think I'll manage now :)

**Reply by janaobsteter** [reply]
Just another question - the `".node:not(.leaf) > .sym {visibility: hidden}" ` doesn't seem to be working (I've tried a couple of other variations from the tutorials). Any idea why?

This is my code:
```
yellowtree = ts10.at(11506337)
styles = []
# Create a style for each population, programmatically (or just type the string by hand)
for colour, p in zip(["#ad2793", "#228c8d", "#feb72d", "#a8db34", "#f07f4f", "#f7e225","#310597"], ts10.populations()):
    # target the symbols only (class "sym")
    s = f".node.p{p.id} > .sym " + "{" + f"fill: {colour}" + "}"
    styles.append(s)
    print(f'"{s}" applies to nodes from population {json.loads(p.metadata)["subspecies"]} (id {p.id})')
styles.append(".node:not(.leaf) > .sym {visibility: hidden}")
css_string = " ".join(styles)

yellowtree.draw_svg(
    path="/home/jana/EddieDir/PCAYellow_SVG.svg",
    size=(700, 400),
    node_labels={},    # Remove all node labels for a clearer viz
    style=css_string,
    symbol_size=8 # Apply the stylesheet
)
```

And the figure is:
![image](https://user-images.githubusercontent.com/18226551/176744103-6023f4c5-f19f-42eb-b811-8a71953ae319.png)

**Reply by hyanwong** [reply]
Hmm, this works for me. How are you viewing the file? If you look at it in chrome it might work. Sometime Inkscape doesn't like stylesheets:

```
# works in a Jupyter Notebook
import msprime
import tskit
from IPython.display import SVG

deme_size = 500  # Population size of each deme
num_demes = 3
num_deme_samples = 25

demography = msprime.Demography.stepping_stone_model(
    [deme_size] * num_demes,
    migration_rate=0.05
)

mu = 1e-8 # Human-like mutation rate

ts = msprime.sim_ancestry(
    {population_id: num_deme_samples for population_id in range(num_demes)},
    sequence_length=5e6, # 5Mb
    demography=demography,
    recombination_rate=1e-8, # Human-like recombination rate
    random_seed=123,
)

ts10 = msprime.sim_mutations(
    ts,
    rate=mu,
    random_seed=321
)

yellowtree = ts10.at(1e6)
styles = []
# Create a style for each population, programmatically (or just type the string by hand)
for colour, p in zip(["#ad2793", "#228c8d", "#feb72d", "#a8db34", "#f07f4f", "#f7e225","#310597"], ts10.populations()):
    # target the symbols only (class "sym")
    s = f".node.p{p.id} > .sym " + "{" + f"fill: {colour}" + "}"
    styles.append(s)

styles.append(".node:not(.leaf) > .sym {visibility: hidden}")
css_string = " ".join(styles)

SVG(yellowtree.draw_svg(
    #path="/home/jana/EddieDir/PCAYellow_SVG.svg",
    size=(700, 400),
    node_labels={},    # Remove all node labels for a clearer viz
    style=css_string,
    symbol_size=8 # Apply the stylesheet
))
```
<img width="791" alt="Screenshot 2022-06-30 at 22 52 02" src="https://user-images.githubusercontent.com/4699014/176785065-6231ef93-9bee-4f2c-98c4-6e7730399b93.png">

**Reply by hyanwong** [reply]
Maybe whatever SVG program you are using doesn't like the ":not" selector. You could resort to the chromium hack: https://tskit.dev/tutorials/viz.html#sec-tskit-viz-converting-note

**Reply by janaobsteter** [reply]
Argh, I was opening it in Inkscape. Never would have of that being a problem, thanks!

**Reply by hyanwong** [reply]
Yeah, Inkscape support for some of the CSS is annoyingly patchy.

**Reply by jeromekelleher** [reply]
Maybe worth a "warning" admonition in the docs @hyanwong?

**Reply by hyanwong** [reply]
It looks like Inkscape doesn't support the "visibility" property. I could just change that to "display: none" in the example instead.
